pkg/runtime/mac/mac.go

```diff
@@ -0,0 +1,159 @@
+	}
+
+	_, err = m.cliClient.Create(imageID)
+	if err != nil {
+		return api.MakeFailedUpdateStatus(unit.Name, unit.Image, "VMCreateFailed"), err
+	}
+	return api.MakeStillCreatingStatus(unit.Name, unit.Image, "VMTemplatePulled"), nil
+}
+
+func (m *MacRuntime) StartContainer(unit api.Unit, spec *api.PodSpec, podName string) (*api.UnitStatus, error) {
+	vmIDInterface, ok := m.UnitsVMIDs.Load(unit.Name)
+	if !ok {
+		return nil, fmt.Errorf("cannot find vm id for unit %s", unit.Name)
+	}
+	vmID := vmIDInterface.(string)
+	unitStatus, err := m.cliClient.Start(vmID)
+	if err != nil {
+		return nil, err
+	}
+	if unitStatus.Status != AnkaStatusOK {
+		return api.MakeFailedUpdateStatus(unit.Name, unit.Image, "VMStartFailed"), fmt.Errorf("cannot start vm: %s", unitStatus.Message)
+	}
+	started := true
+	return &api.UnitStatus{
+		Name: unit.Name,
+		State: api.UnitState{
+			Running: &api.UnitStateRunning{StartedAt: api.Now()},
+		},
+		Image:   unit.Image,
+		Ready:   true,
+		Started: &started,
+	}, nil
+}
+
+func getVMImageIDFromUnitImage(unit api.Unit) (string, error) {
+	// Let's set a convention that if container image has prefix mac-anka-img:
+	// then we'll treat suffix as anka VM Image ID.
+	// e.g. mac-anka-img:<vm-id>
+	vmImgName := strings.Split(unit.Image, ":")
+	if len(vmImgName) < 2 {
+		return "", fmt.Errorf("image has to be in format %s:vm-id", vmImagePrefix)
+	}
+	if vmImgName[0] != vmImagePrefix {
+		return "", fmt.Errorf("image has to be in format %s:vm-id", vmImagePrefix)
+	}
+	return vmImgName[1], nil
+}
+
+func (m *MacRuntime) RemoveContainer(unit *api.Unit) error {
+	// TODO
+	return nil
+}
+
+func (m *MacRuntime) ContainerStatus(unitName, unitImage string) (*api.UnitStatus, error) {
+	// TODO
+	vmIDinterface, ok := m.UnitsVMIDs.Load(unitName)
+	if !ok {
+		return nil, fmt.Errorf("cannot find vm id for unit %s", unitName)
+	}
+	vmID := vmIDinterface.(string)
+	showOutput, err := m.cliClient.Show(vmID)
+	if err != nil {
+		return nil, err
+	}
+	switch showOutput.Body.Status {
+	case "stopped":
+		// vm is stopped
+		return api.MakeFailedUpdateStatus(unitName, unitImage, "VMStopped"), nil
+	case "suspended":
+		// vm is suspended
+		return api.MakeFailedUpdateStatus(unitName, unitImage, "VMSuspended"), nil
+	case "running":
+		return &api.UnitStatus{
+			Name:  unitName,
+			State: api.UnitState{Running: &api.UnitStateRunning{}},
+			Image: unitImage,
+		}, nil
+	case "failed":
+		return api.MakeFailedUpdateStatus(unitName, unitImage, "VMFailed"), nil
+
+	}
+	return nil, nil
+}
+
+func (m *MacRuntime) GetLogBuffer(unitName string) (*logbuf.LogBuffer, error) {
+	return nil, nil
+}
+
+func (m *MacRuntime) ReadLogBuffer(unitName string, n int) ([]logbuf.LogEntry, error) {
+	return nil, nil
+}
+
+func (m *MacRuntime) UnitRunning(unitName string) bool {
+	return true
+}
+
+func (m *MacRuntime) GetPid(unitName string) (int, bool) {
+	return 0, false
+}
+
+func (m *MacRuntime) SetPodNetwork(netNS, podIP string) {
+	return
+}
```

Comment:
I think we'll have to rethink how a pod IP is allocated. With the old runtime, we allocate an extra IP to the main ENI, and route that into the network namespace. That's a special case, though, not something we can do with other runtimes.